src/komet/kdist/soroban-semantics/host/vector.md

```diff
@@ -24,6 +24,86 @@ module HOST-VECTOR
         <locals> .Map </locals>
 ```
 
+## vec_get
+
+```k
+    rule [hostfun-vec-get]:
+        <instrs> hostCall ( "v" , "1" , [ i64  i64  .ValTypes ] -> [ i64  .ValTypes ] )
+              => loadObject(HostVal(INDEX))
+              ~> loadObject(HostVal(VEC))
+              ~> vecGet
+                 ...
+        </instrs>
+        <locals>
+          0 |-> < i64 > VEC
+          1 |-> < i64 > INDEX
+        </locals>
+
+    syntax InternalInstr ::= "vecGet"   [symbol(vecGet)]
+ // ----------------------------------------------------
+    rule [vecGet]:
+        <instrs> vecGet => VEC {{ I }} orDefault HostVal(0) ... </instrs> // use 'orDefault' to avoid non-total list lookup
```

Comment:
@bbyalcinkaya could you please elaborate on how `orDefault` operates here? And how it is used to avoid a scenario of non-total list lookup? I'm not entirely familiar with this production, so if you have a reference for me to take a deeper look into it, it would be more than enough.

Reply:
It is a typed and total list lookup defined [here](https://github.com/runtimeverification/komet/blob/fxdao-hostcalls/src/komet/kdist/soroban-semantics/data.md?plain=1#L183). It returns the default value if the index is out of bounds or the list item at that index is not a `HostVal`. 

I could've used the normal list lookup but that would require:
1. projection cast on the list lookup result because `_ [_]` returns `KItem`: `{VEC[I]}:>HostVal`
2. side condition to check the list item sort: `isHostVal(VEC[I])`
3. `preserves-definedness` attribute to avoid booster backend fallbacks (and manually check if it actually does)